Increase coverage for SAMRecord query comparators

magicDGS opened the pull request:
### Description

* Testing explicitly SAMRecordQueryNameComparator
* Adding tests for SAMRecordQueryHashComparator

### Checklist

- [x] Code compiles correctly
- [x] New tests covering changes and new functionality
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

yfarjoun commented on src/test/java/htsjdk/samtools/SAMRecordQueryHashComparatorTest.java:
```diff
@@ -0,0 +1,68 @@
+        final SAMRecord b = new SAMRecord(null);
+        b.setReadName("B");
+        // hashes are providing a different order in this case
+        Assert.assertTrue(COMPARATOR.compare(a, b) > 0);
```
why should the order be as such? trial and error, no? better to check that they are `!=` since the hash function might change.

yfarjoun replied:
I saw this while reviewing your code...could you fix the comment here: htsjdk/samtools/SAMRecordQueryHashComparator.java:58 ?

magicDGS replied:
Done (using `|=`). What should be fixed in SAMRecordQueryHashComparator.java:58?

yfarjoun replied:
well, the comment is wrong since the fallback in that case is to the file order...but this isn't your fault...feel free to ignore.

yfarjoun commented on src/test/java/htsjdk/samtools/SAMRecordQueryHashComparatorTest.java:
```diff
@@ -0,0 +1,68 @@
+    @Test(dataProvider = "equalNameComparisonData", dataProviderClass = SAMRecordQueryNameComparatorTest.class)
+    public void testCompareEqualNames(final SAMRecord record1, final SAMRecord record2, final int sign) throws Exception {
+        final int comparisonResult = COMPARATOR.compare(record1, record2);
+        switch (sign) {
```
could you use 
```java
Assert.assertEquals(Integer.signum(comparisonResult),sign)
```
instead of that switch statement?

magicDGS replied:
Sure, I didn't know the signum method!

yfarjoun commented on src/test/java/htsjdk/samtools/SAMRecordQueryNameComparatorTest.java:
```diff
@@ -0,0 +1,136 @@
+    @Test(dataProvider = "equalNameComparisonData")
+    public void testCompareEqualNames(final SAMRecord record1, final SAMRecord record2, final int sign) throws Exception {
+        final int comparisonResult = COMPARATOR.compare(record1, record2);
+        switch (sign) {
```
here as well (clean up switch)

magicDGS replied:
Done!

magicDGS commented:
I addressed all your comments except one, @yfarjoun, because I don't know what should I fix in the comment for `SAMRecordQueryHashComparator.java:58`.

Back to you @yfarjoun!

yfarjoun commented:
1. Add a newline to the end of the files so that github removes that angry red symbol.
2. wait till tests pass
3. merge 👍 
4. thanks!

magicDGS commented:
Added new line. Thanks for reviewing!
